### PR TITLE
Fix(ngx_multi_upstream_module): fake request isn't freed

### PR DESCRIPTION
**Pull request by spacewander**

### CODE REVIEW

File: modules/ngx_multi_upstream_module/ngx_http_multi_upstream_module.c [hunk 1]
```diff
@@ -370,6 +372,23 @@ ngx_http_multi_upstream_add_data(ngx_connection_t *c, ngx_http_request_t *r)
...
+static void
+ngx_http_multi_upstream_free_fake_request(void *data)
+{
+    ngx_http_request_t  *r = data;
+    ngx_pool_t          *pool;
+
+    ngx_log_debug1(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
+                   "multi upstream fake request cleanup: %p", r);
+
+    pool = r->pool;
```
Comment (MengqiWu) [inline modules/ngx_multi_upstream_module/ngx_http_multi_upstream_module.c:384]: we do like this first, but will make some mistake for stub_status module, so we use like this after:
    fake_r->logged = 1;
    ngx_http_free_request(fake_r, 0);

**Review by MengqiWu — CHANGES_REQUESTED** [review]

File: modules/ngx_multi_upstream_module/ngx_http_multi_upstream_module.c (hunk 1 above)
Comment (spacewander) [reply modules/ngx_multi_upstream_module/ngx_http_multi_upstream_module.c:384]: I can't find `ngx_http_free_request(fake_r, 0)` in the source code of Tengine?

Comment (spacewander) [reply modules/ngx_multi_upstream_module/ngx_http_multi_upstream_module.c:384]: Or do you mean changing the `destroy r->pool` to
```c
fake_r->logged = 1;
ngx_http_free_request(fake_r, 0);
```

Comment (MengqiWu) [reply modules/ngx_multi_upstream_module/ngx_http_multi_upstream_module.c:384]: yes, i mean changing the destroy r->pool to like this
fake_r->logged = 1;
ngx_http_free_request(fake_r, 0);

and，i mean we fix this on alibaba internal use destroy  r->pool , and we encounter mistake on sub_stat, so we use like this after, so i suggest you change to use ngx_http_free_request and set fake_r->logged = 1;

Comment (spacewander) [reply modules/ngx_multi_upstream_module/ngx_http_multi_upstream_module.c]: @MengqiWu 
Solved in the new commit.

**Review by MengqiWu — APPROVED** [review]

**Pull request merged**